code_examples/sdk_examples/src/core_features/signCallback/useEncryptionSignCallback.ts

```diff
@@ -0,0 +1,30 @@
+import * as Kilt from '@kiltprotocol/sdk-js'
+import { naclSeal } from '@polkadot/util-crypto'
+
+export function useEncryptionSignCallback({
+  keyAgreement,
+  didDocument
+}: {
+  keyAgreement: Kilt.KiltEncryptionKeypair
+  didDocument: Kilt.DidDocument
+}): Kilt.EncryptCallback {
+  return async function encryptCallback({
```

Comment:
What's the advantage of having a function that simply wraps around another function? Can't these two be flattened into a single one?

Reply:
This is a callback function. Without passing these parameters, you can't get that information without wrapping it in another function.

Reply:
Yes what I am suggesting is to return the callback and not the function. Something like this:

```ts
export function useEncryptionCallback({
  keyAgreement,
  keyAgreementUri
}: {
  keyAgreement: Kilt.KiltEncryptionKeypair
  keyAgreementUri: Kilt.DidResourceUri
}): Kilt.EncryptCallback {
  return (async ({data, peerPublicKey}): Promise<Kilt.EncryptResponseData> => {
    const { sealed, nonce } = naclSeal(
      data,
      keyAgreement.secretKey,
      peerPublicKey
    )
    return {
      nonce,
      data: sealed,
      keyUri: keyAgreementUri
    }
  })
}
```

It's more of a stylistic matter, I guess. It returns a callback, so it should actually return a callback.

Reply:
I think we should discuss how we want to present these callbacks. At the moment, if I look through the current implementation, it differs from my solution. It also differs from your solution.

We should decide to remain constant or provide multiple ways of doing it.

Reply:
For me, we should stick to one way of doing it. All our stuff at the moment is complicated enough without having different styles throw into the mix.

Reply:
I mistakely put that comment into the wrong comment :X

Reply:
So do you at least agree that having a nested function inside a function to just return a callback is unnecessary?

Reply:
That is exactly what your function does above anyway.